### PR TITLE
Migrate DagFileProcessorManager.clear_nonexistent_import_errors to internal API

### LINKED ISSUE #28785: AIP-44 Migrate DagFileProcessorManager.clear_nonexistent_import_errors to Internal API
https://github.com/apache/airflow/blob/main/airflow/dag_processing/manager.py#L773

### PR DESCRIPTION
**Pull request by snjypl**
Fixes: #28785 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/dag_processing/manager.py [hunk 1]
```diff
@@ -725,7 +726,7 @@ def _refresh_dag_dir(self):
 
             try:
                 self.log.debug("Removing old import errors")
-                self.clear_nonexistent_import_errors()
+                self.clear_nonexistent_import_errors(file_paths=self._file_paths)
```
Comment (vincbeck) [inline airflow/dag_processing/manager.py]: It is a static method now

```suggestion
                DagFileProcessorManager.clear_nonexistent_import_errors(file_paths=self._file_paths)
```

File: airflow/dag_processing/manager.py [hunk 2]
```diff
@@ -769,16 +770,19 @@ def _print_stat(self):
                 self._log_file_processing_stats(self._file_paths)
             self.last_stat_print_time = time.monotonic()
 
+    @staticmethod
+    @internal_api_call
     @provide_session
-    def clear_nonexistent_import_errors(self, session):
+    def clear_nonexistent_import_errors(file_paths: list[str] = [], session=NEW_SESSION):
```
Comment (vincbeck) [inline airflow/dag_processing/manager.py]: I dont think the default value for `file_paths` is needed

```suggestion
    def clear_nonexistent_import_errors(file_paths: list[str] | None, session=NEW_SESSION):
```

**Review by vincbeck — CHANGES_REQUESTED** [review]

**Comment by snjypl** [comment]
@vincbeck  thanks for the review. i have made the suggested  changes.

**Review by vincbeck — APPROVED** [review]

**Comment by vincbeck** [comment]
@mhenc

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
conflicts to resolve.

**Review by potiuk — APPROVED** [review]

**Pull request merged**